torch/compiler/__init__.py

```diff
@@ -461,3 +461,68 @@ def load_cache_artifacts(serialized_artifacts: bytes) -> Optional["CacheInfo"]:
     from ._cache import CacheArtifactManager, CacheInfo
 
     return CacheArtifactManager.deserialize(serialized_artifacts)
+
+
+if TYPE_CHECKING:
+    from torch._fullgraph import _Package
+
+_FULLGRAPH_PACKAGE: Optional["_Package"] = None
+
+
+# NOTE: [Fullgraph Package] This is an experimental feature that allows users to
+#       explicitly package precompiled artifacts into a single file.
+#       TODO Eventually we should come up with a context manager style API. To
+#       reduce the complexity of landing changes, we first introduce a set of
+#       stateful interfaces as the future building blocks to begin with.
```

Comment:
This is going to be intended to be a real API right? Do you think it's too early to put our best foot forward on the API (which includes not having underscores?)